hub/integrations/pytorch.py

```diff
@@ -81,7 +81,6 @@ def dataset_to_pytorch(
     collate_fn: Optional[Callable] = None,
     pin_memory: Optional[bool] = False,
 ):
-    dataset.flush()
```

Comment:
can we actually add a try catch block for this or a check to flush if not in read mode? 
In case somebody sends a dataset to pytorch within 'with' syntax while writing, it's good to flush the data before sending it to pytorch otherwise the cache data will not be accessed

Reply:
this is not a good way to solve this problem.

Reply:
there's probably a better way to solve this that is more robust, but for the time being sure